catalog/resourcemanagement/resourcemanagement-usage-ui/src/main/java/org/codice/ddf/resourcemanagement/usage/service/DataUsage.java

```diff
@@ -0,0 +1,225 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataUsage extends RouteBuilder implements DataUsageMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataUsage.class);
+
+    // 23:30 Daily
+    public static final String DEFAULT_CRON_TIME = "0+30+23+*+*+?";
+
+    public static final String CRON_TIME_ID_KEY = "data-usage-cron-key";
+
+    public static final String CRON_TIME_KEY = "cron_time";
+
+    public static final String ID = "id";
+
+    public static final String TXT_PREFIX = "_txt";
+
+    private static final String PREFERENCES_TYPE = "preferences";
+    
+    private static final String LNG_PREFIX = "_lng";
+
+    private AttributesStore attributesStore;
+
+    private PersistentStore persistentStore;
+
+    private String cronTime;
+
+    private ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+
+    public DataUsage() {
+        registerMbean();
+    }
+
+    public void init() {
+        this.cronTime = getPersistentCronTime();
+        LOGGER.debug("Set Cron Time as : {}", this.cronTime);
+    }
+
+    private String getPersistentCronTime() {
+        try {
+            readWriteLock.readLock()
+                    .lock();
+            List<Map<String, Object>> mapList = persistentStore.get(PREFERENCES_TYPE);
+            readWriteLock.readLock()
+                    .unlock();
+
+            for (Map<String, Object> preference : mapList) {
+                String id = (String)preference.get(ID + TXT_PREFIX);
+                if (id != null && id.equals(CRON_TIME_ID_KEY)) {
+                    return (String) preference.get(CRON_TIME_KEY + TXT_PREFIX);
+                }
+            }
+        } catch (PersistenceException e) {
+            readWriteLock.readLock()
+                    .unlock();
+            return DEFAULT_CRON_TIME;
+        }
+        return DEFAULT_CRON_TIME;
+    }
+
+    private void setPersistentCronTime() {
+        PersistentItem persistentItem = new PersistentItem();
+        persistentItem.addIdProperty(CRON_TIME_ID_KEY);
+        persistentItem.addProperty(CRON_TIME_KEY, this.cronTime);
+
+        try {
+            readWriteLock.readLock()
+                    .lock();
+            persistentStore.add(PREFERENCES_TYPE, persistentItem);
+
+        } catch (PersistenceException e) {
+            LOGGER.error("Error adding Cron Time to Persistent Store.");
+        } finally {
+            readWriteLock.readLock()
+                    .unlock();
+        }
+    }
+
+    private void registerMbean() {
+        ObjectName objectName = null;
+        MBeanServer mBeanServer = null;
+        try {
+            objectName = new ObjectName(DataUsage.class.getName() + ":service=datausage");
+            mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        } catch (MalformedObjectNameException e) {
+            LOGGER.error("Unable to create Data Usage Configuration MBean.", e);
+        }
+        if (mBeanServer != null) {
```

Comment:
Do we not want to do anything if the mbeanserver is null? right now it would just pass through.
also, Personally I always like the reduce cyclomatic complexity route making this a `if (mBeanServer == null)..` and then doing a return or whatever we want to do in the case it is still null.